src/baseTask.ts

```diff
@@ -0,0 +1,12 @@
+// BaseTask.ts
+class BaseTask {
+    condition(inputData: any): boolean {
+        return true; // Default condition, always return true
+    }
+
+    async execute(inputData: any): Promise<any> {
+        console.log('Base task execution with data:', inputData);
+    }
+}
+
+export default BaseTask;
```

Comment:
lets make class and function as abstract.  no need to implement. As we will never create an object for this.